stock_custom_exchange_rate_date/models/stock_move.py

```diff
@@ -18,4 +18,6 @@ def _get_price_unit(self):
         price_unit = line.currency_id._convert(
             price_unit, line.company_id.currency_id, line.company_id, self.picking_id.exchange_rate_date, round=False
         )
-        return price_unit
+        if self.product_id.lot_valuated:
+            return {lot: price_unit for lot in self.lot_ids}
+        return {self.env["stock.lot"]: price_unit}
```

Comment:
Shorter and consistend with the native method:

`return dict.fromkeys(self.lot_ids, price_unit)`

Reply:
This does not work if ```self.lot_ids``` is empty as it returns an equally empty dictionary.

Reply:
Sorry, I meant in the above line.

Reply:
Done